app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+    def new 
+    end 
+
+    def create 
+        user = User.find_by(params[:id])
+        # binding.pry
+        session[:user_id] = user.id
+        if user.role == "admin" 
+            flash[:notice] = "You are logged in as #{user.name}"
+            redirect_to "/admin/dashboard"
+        elsif user.role == "merchant"
+            flash[:notice] = "You are logged in as #{user.name}"
+            redirect_to "/merchant/dashboard"
+        else 
+            flash[:notice] = "You are logged in as #{user.name}"
+            redirect_to "/profile"
```

Comment:
Based on their role, user gets redirected to a different route. Admin and Merchant routes are namespaced.

Reply:
ya, not sure if we want to name space the merchant, I think it works as is though, but I will play around with it for a bit